### PR TITLE
PR with squashed and rebased pms-ni commits on master

### LINKED ISSUE #72: Reverse order of arguments for `buildmodel!`
It should be `buildmodel!(operation_model, sys)` and not `buildmodel!(sys, operation_model)`.

### LINKED ISSUE #73: PM integration problem with expressions errors
I have moved forward with the PowerModels integration through expressions. I am getting stuck in this error

```
MethodError: no method matching getindex(::JuMP.GenericAffExpr{Float64,VariableRef}, ::Int64)

Stacktrace:
 [1] ref(::GenericPowerModel{PowerSimulations.DCAngleForm}, ::Int64, ::Symbol, ::Int64, ::String, ::Int64) at /Users/jdlara/.julia/packages/PowerModels/mNbep/src/core/base.jl:150
 [2] #constraint_kcl_ni_expr#172(::Int64, ::Int64, ::Function, ::GenericPowerModel{PowerSimulations.DCAngleForm}, ::Int64) at /Users/jdlara/.julia/dev/PowerSimulations/src/network_models/powermodels_balance.jl:166
 [3] (::getfield(PowerSimulations, Symbol("#kw##constraint_kcl_ni_expr")))(::NamedTuple{(:nw,),Tuple{Int64}}, ::typeof(PowerSimulations.constraint_kcl_ni_expr), ::GenericPowerModel{PowerSimulations.DCAngleForm}, ::Int64) at ./none:0
 [4] post_nip_expr(::GenericPowerModel{PowerSimulations.DCAngleForm}) at /Users/jdlara/.julia/dev/PowerSimulations/src/network_models/powermodels_balance.jl:76
 [5] #build_generic_model#126(::Bool, ::Bool, ::Base.Iterators.Pairs{Symbol,Model,Tuple{Symbol},NamedTuple{(:jump_model,),Tuple{Model}}}, ::Function, ::Dict{String,Any}, ::getfield(Main, Symbol("##11#13")), ::typeof(PowerSimulations.post_nip_expr)) at /Users/jdlara/.julia/packages/PowerModels/mNbep/src/core/base.jl:258
 [6] (::getfield(PowerModels, Symbol("#kw##build_generic_model")))(::NamedTuple{(:multinetwork, :jump_model),Tuple{Bool,Model}}, ::typeof(build_generic_model), ::Dict{String,Any}, ::Function, ::Function) at ./none:0
 [7] #build_nip_model#166 at /Users/jdlara/.julia/dev/PowerSimulations/src/network_models/powermodels_balance.jl:19 [inlined]
 [8] (::getfield(PowerSimulations, Symbol("#kw##build_nip_model")))(::NamedTuple{(:jump_model,),Tuple{Model}}, ::typeof(PowerSimulations.build_nip_model), ::Dict{String,Any}, ::Function) at ./none:0
 [9] top-level scope at In[8]:2
```

I can't make a lot of sense of the error because it points to a line in `PowerModels` related to multiconductor. This code can reproduce the error: 

```
using PowerSystems
base_dir = dirname(dirname(pathof(PowerSystems)))
include(joinpath(base_dir,"data/data_5bus.jl"));
sys5 = PowerSystem(nodes5, generators5, loads5_DA, branches5, nothing,  100.0);
using PowerModels
const PM = PowerModels
using PowerSimulations
const PS = PowerSimulations
using Ipopt
using JuMP
ipopt_optimizer = with_optimizer(Ipopt.Optimizer)
Net = PS.DCAngleForm
m = Model(with_optimizer(Ipopt.Optimizer));
netinjection = PS.instantiate_network(Net, sys5);
PS.constructdevice!(m, netinjection, ThermalGen, PS.ThermalDispatch, Net, sys5);
#Here goes the network construction functions
PS.nodalflowbalance(m, netinjection, Net, sys5)
PM_F = (data::Dict{String,Any}; kwargs...) -> PM.GenericPowerModel(data, Net; kwargs...)
PM_object = PS.build_nip_model(m.ext[:PM_object], PM_F, jump_model=m);
# 
@objective(m, Min, m.obj_dict[:objective_function])
```

FYI @ccoffrin

### PR DESCRIPTION
**Pull request by kdheepak**
- wip, basic nip powermodels formulation
- update powermodels verison
- fix powermodel building bugs
- adding tests for build_nip_model
- fix tabbing
- adding note on bounding of net_injection variables
- Add translator and testing WIP
- add nw capability in translator
- add support for multinetwork data
- Modifications to Network Constructor
- add InfrastructureModels to manifest
- fix to multinetwork constraint_voltage
- restore socwr test
- minor fix to network types
- Add integration with PM
- add start to pwl
- Enable external definition of system_formulation
- update testing
- Fix test errors in NetFlow
- remove failing tests
- add start to flows
- enable AC PF
- add transformer support for the AC power flow
- Update Manifest
- wip, draft of build_nip_expr_model
- add conductor to constraint_kcl_ni_expr template
- update Manifest
- update Manifest
- Update PM Integration
- update Manifest
- reorder PowerSiimulations.jl
- Integration with PM WIP.
- Fixes #72
- Fixes #73
- add complete network construction
- add testing for PM integration
- update network constructor
- remove pu conversion
- reverse change in the post_method
- reverse JuMP version in Manifest
- Fix testing
- remove matpower data folder
- add test for PM.QCWRForm
- fix ramping constraint generation

### CODE REVIEW

**Comment by kdheepak** [comment]
Manifest.toml has issues because of the merge conflicts, and need to be fixed.

**Comment by kdheepak** [comment]
See #74 for individual commit changes.

File: src/device_models/common/variable_cost.jl [hunk 1]
```diff
@@ -22,7 +22,7 @@ end
 
 function pwlgencost(m::JuMP.Model, variable::VariableRef, cost_component::Array{Tuple{Float64, Float64}})
 
-    pwlvars = @variable(m, [i = 1:(length(cost_component)-1)], base_name = "pwl_{$(variable)}")
+    pwlvars = @variable(m, [i = 1:(length(cost_component)-1)], base_name = "pwl_{$(variable)}", start = 0.0, lower_bound = 0.0, upper_bound = (cost_component[i+1][1] - cost_component[i][1]))
```
Comment (kdheepak) [inline src/device_models/common/variable_cost.jl]: See #79 

cc @claytonpbarrows

Comment (claytonpbarrows) [reply src/device_models/common/variable_cost.jl]: I think this can't have the bounds. I'm not exactly sure why, but it creates infeasible problems.

**Pull request merged**